runtime/signals_nat.c

```diff
@@ -89,6 +89,18 @@ void caml_garbage_collection(void)
                             nallocs, alloc_len);
 }
 
+void caml_notify_action()
+{
+  Caml_state->young_limit = Caml_state->young_alloc_end;
+  /* Some ports cache [Caml_state->young_limit] in a register.
+     Use the signal context to modify that register too, but only if
+     we are inside OCaml code (not inside C code). */
+#if defined(CONTEXT_PC) && defined(CONTEXT_YOUNG_LIMIT)
+  if (caml_find_code_fragment_by_pc((char *) CONTEXT_PC) != NULL)
+    CONTEXT_YOUNG_LIMIT = (context_reg) Caml_state->young_limit;
+#endif
```

Comment:
On the one hand, the caching of `young_limit` in a register is going away with #9876, so the highlighted code will go away.

On the other hand, this code was wrong to begin with.  `CONTEXT_PC` and `CONTEXT_YOUNG_LIMIT` can only be used from a signal handler defined using `DECLARE_SIGNAL_HANDLER`, because the latter refer to function parameters of the former.

Reply:
Thanks for the clarification; this code should be lifted out of the function and back into the callsite where it was used.

Reply:
Yes, I realised that this old code of mine was fishy, but it was meant to disappear with that other PR so I did not pay more attention to it. Sorry for this unnecessary (and wrong) commit.

Reply:
I removed the erroneous fix.